src/App.js

```diff
@@ -1,53 +1,145 @@
 import { useEffect, useState } from 'react';
 import './App.css';
 import ProfileCard from './components/profile-card';
-import profileData from './mock-data';
-import {message} from 'antd'
+import {message, notification, Spin, Tag, Descriptions, Row} from 'antd'
 import _ from 'lodash'
+import { getProfilesData } from './network' 
+
+function mapProfileData(rawData){
+  return rawData.map(({name, email, dob, picture}) => ({
+    name: name.first + ' ' + name.last,
+    email,
+    age: dob.age,
+    imgUrl: picture.large
+  }))
+}
+let remainingCardsNum;
+const REMAINING_CARD_THRESHOLD = 2;
 
 function App() {
   const [profiles, setProfiles] = useState([]);
   const [currentProfileIndex, setProfileIndex] = useState(0);
+  const [viewedProfiles, setViewedProfiles] = useState([]);
+  const [shouldPrefetchProfile, setShouldPrefetchProfile] = useState(false);
+  const [isLoading, setLoading] = useState(false)
```

Comment:
there are too many states to keep track of. if there's a bug in manually setting `isLoading`, `shouldPrefetchProfile` or `currentProfileIndex` somewhere, not to mention the extra variable `remainingCardsNum`. it's tricky to hold all that info in our head when debugging.

this also makes it challenging for the reader/reviewer to focus on the approach and find the simplicity. the only thing you should need is a `userList` and a running index.. wdyt?

Reply:
Good catch! I agree with you and hence this is the commit to clean up all redundant states & rename vars for comprehension [state cleanup](https://github.com/SophiePham92/react-tinder-cards/pull/5/commits/f8fb4a19ff022bc89d50cf307a77b3260ed00584) 😄 
I want to do it fast at first, so the solution was messy. When everything seems to work, I come back for revision & clean up. You welcome to read the new version 🍺